File: server/router/cloudpods/baseRes/host.go
package baseRes

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type BaseHostRouter struct {
}

func (s *BaseHostRouter) InitBaseHostRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/baseRes/baseHost")
	vpcApi := v1.ApiGroupApp.CloudApiGroup.BaseResApiGroup.BaseHostApi

	{
		dataRouter.POST("getBaseHostList", vpcApi.GetBaseHostList)                     // 获取宿主机列表
		dataRouter.POST("getBaseBareHostList", vpcApi.GetBaseBareHostList)             // 获取物理机列表
		dataRouter.POST("getImpiInfo", vpcApi.GetImpiInfo)                             // 获取IMPI信息
		dataRouter.POST("getLoginInfo", vpcApi.GetLoginInfo)                           // 获取物理机登录信息
		dataRouter.POST("updateBareHostStatus", vpcApi.UpdateBareHostStatus)           // 修改物理机状态
		dataRouter.POST("updateBareHostDescription", vpcApi.UpdateBareHostDescription) // 修改物理机备注
		dataRouter.POST("deleteBareHost", vpcApi.DeleteBareHost)                       // 删除物理机
		dataRouter.POST("deleteBareHostByIds", vpcApi.DeleteBareHostByIds)             // 批量删除物理机
		dataRouter.POST("addBareHost", vpcApi.AddBareHost)                             // 创建物理机
		dataRouter.POST("getWebConsoleSSH", vpcApi.GetWebConsoleSSH)                   // 远程连接物理机

		dataRouter.POST("queryCpu", vpcApi.GetHostMonitorCPUData)             // 获取监控CPU数据
		dataRouter.POST("queryMem", vpcApi.GetHostMonitorMemData)             // 获取监控内存数据
		dataRouter.POST("queryDisk", vpcApi.GetHostMonitorDiskData)           // 获取监控磁盘使用率
		dataRouter.POST("queryDiskRead", vpcApi.GetHostMonitorDiskReadData)   // 获取监控磁盘读速率
		dataRouter.POST("queryDiskWrite", vpcApi.GetHostMonitorDiskWriteData) // 获取监控磁盘写速率
		dataRouter.POST("queryBpsRecv", vpcApi.GetHostMonitorBpsRecvData)     // 获取监控网络入带宽
		dataRouter.POST("queryBpsSent", vpcApi.GetHostMonitorBpsSentData)     // 获取监控网络出带宽
	}
}

File: web/src/api/cloudpods/baseRes/host.js
import request from "@/utils/request"

//获取宿主机列表
export function getBaseHostList(params) {
    return request({
        url: 'cloud/baseRes/baseHost/getBaseHostList',
        method: 'post',
        data: params
    })
}
//获取物理机列表
export function getBaseBareHostList(params) {
    return request({
        url: 'cloud/baseRes/baseHost/getBaseBareHostList',
        method: 'post',
        data: params
    })
}
//获取IMPI信息
export function getImpiInfo(params) {
    return request({
        url: 'cloud/baseRes/baseHost/getImpiInfo',
        method: 'post',
        data: params
    })
}
//获取物理机登录信息
export function getLoginInfo(params) {
    return request({
        url: 'cloud/baseRes/baseHost/getLoginInfo',
        method: 'post',
        data: params
    })
}
//修改物理机状态
export function updateBareHostStatus(params) {
    return request({
        url: 'cloud/baseRes/baseHost/updateBareHostStatus',
        method: 'post',
        data: params
    })
}
//修改物理机备注
export function updateBareHostDescription(params) {
    return request({
        url: 'cloud/baseRes/baseHost/updateBareHostDescription',
        method: 'post',
        data: params
    })
}
//删除物理机
export function deleteBareHost(params) {
    return request({
        url: 'cloud/baseRes/baseHost/deleteBareHost',
        method: 'post',
        data: params
    })
}
//批量删除物理机
export function deleteBareHostByIds(params) {
    return request({
        url: 'cloud/baseRes/baseHost/deleteBareHostByIds',
        method: 'post',
        data: params
    })
}
//新增物理机
export function addBareHost(params) {
    return request({
        url: 'cloud/baseRes/baseHost/addBareHost',
        method: 'post',
        data: params
    })
}
//远程连接物理机
export function getWebConsoleSSH(params) {
    return request({
        url: 'cloud/baseRes/baseHost/getWebConsoleSSH',
        method: 'post',
        data: params
    })
}
//获取CPU监控数据
export  function  getMonitorHostCpuData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryCpu',
        method:'post',
        data: params
    })
}
//获取内存监控数据
export  function  getMonitorHostMemData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryMem',
        method:'post',
        data: params
    })
}
//获取磁盘监控数据
export  function  getMonitorHostDiskData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryDisk',
        method:'post',
        data: params
    })
}
//获取磁盘读速率监控数据
export  function  getMonitorHostDiskReadData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryDiskRead',
        method:'post',
        data: params
    })
}
//获取磁盘写速率监控数据
export  function  getMonitorHostDiskWriteData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryDiskWrite',
        method:'post',
        data: params
    })
}
//获取网络入带宽监控数据
export  function  getMonitorHostBpsRecvData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryBpsRecv',
        method:'post',
        data: params
    })
}
//获取网络出带宽监控数据
export  function  getMonitorHostBpsSentData(params){
    return request({
        url:'cloud/baseRes/baseHost/queryBpsSent',
        method:'post',
        data: params
    })
}